Use short form command line arguments for base64

tvesely opened the pull request:
The busybox version of the `base64` command does not support the long form flag `--decode`, so use the short form `-d` for compatibility.

tvesely committed:
Use short form command line arguments for base64

The busybox version of the base64 command does not support the long
form flag '--decode', so use the short form '-d' for compatibility.

KyleKotowick commented:
Interesting problem. The reason it uses the full-length flag is because apparently `base64`,when installed via Homebrew for MacOS, doesn't support the short `-d` flag ([apparently it means `debug` on Mac](https://electrictoolbox.com/base64-decode-command-line/)).

What it needs is an addition to the script to check if it's Mac, and use `--decode` if so, or `-d` otherwise.

tvesely commented:
Interesting. OK I'll get my hands on a Mac to test this on.

jantari commented:
Coming here after having the same issue in #4, it looks like [some other projects have faced the same macOS issue](https://github.com/traefik/traefik/pull/2344/files).

@tvesely Maybe you can apply the same workaround for macOS they did to your branch here too?

tvesely commented:
I still haven't managed to get my hands on a mac to test, may not get around to this for another few weeks.

Should we close this until I get a chance to work on this?

KyleKotowick commented:
It seems that GitHub Actions supports macOS as a runner, so we could test it there.

If you have the time, if you could write a couple tests, I could set up a workflow that runs them on different operating systems to verify that it works on all.

jantari committed:
Use base64 -d on all Linux platforms and --decode on others

The busybox (Alpine) version of the base64 command does not support the
long form flag '--decode' and the macOS version does not support '-d' so
we have to use different base64 commands depending on the platform.

joncourt commented:
alpine has the same issue. `--decode` is not support only `-d`.

tvesely committed:
Merge pull request #1 from jantari/patch-1

Use base64 -d on all Linux platforms and --decode on others

ovcharenko commented:
Could that PR be merged to celebrate 1 year anniversary?

jantari commented:
Now that would truly be something, we have been stuck on version 0.1.2 of this module forever because of this.

ovcharenko commented:
> Now that would truly be something, we have been stuck on version 0.1.2 of this module forever because of this.

Or upgrade to the latest and use built-in [feature](https://developer.hashicorp.com/terraform/language/resources/terraform-data)

jantari commented:
Ooooh that looks very promising, I'll try it - thank you very much

KyleKotowick commented:
> > Now that would truly be something, we have been stuck on version 0.1.2 of this module forever because of this.
> 
> Or upgrade to the latest and use built-in [feature](https://developer.hashicorp.com/terraform/language/resources/terraform-data)

@ovcharenko The `terraform_data` resource is a replacement for the `null` resource, but unfortunately, it doesn't offer any additional capability. that eliminates the need for this module.

> Could that PR be merged to celebrate 1 year anniversary?

We won't be merging this PR, because we just released a complete rewrite of this module as v0.4.0! It should address all issues with various operating systems and shells, and brings many other improvements.

Please try out the new release and report any issues you find.
